src/main/java/io/github/bobocodebreskul/context/registry/SimpleBeanDefinitionRegistry.java

```diff
@@ -111,40 +144,61 @@ public BeanDefinition getBeanDefinition(String beanName) {
 
   @Override
   public boolean containsBeanDefinition(String beanName) {
+    log.debug("containsBeanDefinition method invoked: beanName={}", beanName);
+    log.info("Checking bean definition registry contains beanName {}", beanName);
     if (isNull(beanName)) {
+      log.error("The specified bean name is null");
       throw new IllegalArgumentException(BEAN_NAME_SHOULD_NOT_BE_NULL);
     }
+    log.trace("Checking bean definition registry contains beanName: {}", beanName);
     return beanDefinitionMap.containsKey(beanName);
   }
 
   @Override
   public Set<String> getBeanDefinitionNames() {
+    log.debug("getBeanDefinitionNames method invoked");
+    log.info("Getting all bean definition names");
+    log.trace("Getting all bean definition names");
```

Comment:
Please choose single level for the same log message, since when end user will setup trace level, he will see duplicated messages for each log level